snapcraft/how-to/quickstart/how-snapcraft-builds-a-snap.md

```diff
@@ -0,0 +1,90 @@
+# How Snapcraft builds snaps
+The Snapcraft tool can perform a variety of operations, including:
+
+* Snap creation operations like build, stage, prime, pack, etc.
+* Store management operations like login, register, sign-build, etc.
+* Extensions-specific operations like expand-extensions and list-extensions.
+
+These operations translate into practical tasks like:
+
+* It can build snaps locally or send remote tasks to Launchpad.
+* It allows the developer to register and log into the Snap Store.
+* It can upload snaps to the Snap Store.
+* It can promote snaps to different channels.
+
+The list of available global options and commands can be checked with one of:
+
+```bash
+snapcraft help
+snapcraft help --all
+```
+
+<h2 id='heading--snapcraft'>snapcraft.yaml</h2>
+
+Snaps are created using a build recipe defined in a file called `snapcraft.yaml`.
+
+When the snapcraft tool is executed on the command line, it will look for the file in the current project work directory, either in the top-level folder or a `snap` subdirectory. If the file is found, snapcraft will then parse its contents and progress the build toward completion.
+
+`snapcraft.yaml` is a configuration file written in the YAML language, with stanzas defining the application structure and behavior. When snapcraft runs and parses this file, it will use the declared information to build the snap. For developers more familiar with the traditional Linux build systems, the process is somewhat similar to the use of a Makefile or an RPM spec file.
+
+You can create the `snapcraft.yaml` file manually, or you can run the `snapcraft init` command to create a template file in the `snap` subdirectory.
+
+```bash
+snapcraft init
+```
+
+A generated template file contains just enough data to build, split across three stanzas:
+
+* Metadata.
+* Confinement level.
+* Build definition.
+
+<h3 id='heading--definitions'>Main definitions inside snapcraft.yaml</h3>
+
+There is no one way for how a snap ought to be assembled. However, most `snapcraft.yaml` files have the same common elements, including a number of mandatory declarations. Below is a short list of these keys, which will be further explained in the Examples sections later in the tutorial.
```

Comment:
As we call out that only *some* keys are mandatory, should we specify in 46-50 which of them are mandatory? Parts and Apps are both optional, for instance - but one wouldn't know it from reading this.